src/main/java/org/spongepowered/api/event/item/inventory/ChangeInventoryEvent.java

```diff
@@ -79,9 +88,58 @@
         /**
          * Fires after an {@link Inventory} transferred an item.
          */
-        interface Post extends Transfer {}
+        interface Post extends Transfer {
+
+        }
 
     }
 
-    interface Pickup extends ChangeInventoryEvent, TargetItemEvent {}
+    /**
+     * Fires after an {@link Item} has been picked up.
+     */
+    interface Pickup extends ChangeInventoryEvent {
+
+        /**
+         * Fires before an {@link Item} is picked up.
+         *
+         * <p>Modifying the picked up items causes this event to be automatically canceled if the inventory does not fit the entire list.</p>
+         */
+        interface Pre extends TargetInventoryEvent, TargetItemEvent, Cancellable {
```

Comment:
Could we do something where it's much like equipment pickup?

Reply:
I was wondering the same thing, my assumption was the Equipment Change events, like when a zombie directly equips armor off the floor.

Reply:
I think Zombie Pickup is not implemented yet